### ISSUE TITLE
Control click not working properly in latest version

### ISSUE THREAD

**Issue by MartinDoyleUK** [theintern/intern#637]
When upgrading from 3.1.1 to 3.2.0 we have a new test failure. We have a simulated control-click using the following test code:

```javascript
this.remote.findByCssSelector("#SR_DISPLAY_NAME .value")
   .pressKeys(keys.CONTROL)
   .click()
   .pressKeys(keys.NULL)
   .end()
```

This is then detected using the following code, triggered by an "ondijitclick" event:

```javascript
var middleButton = evt.button === 2,
   leftButton = evt.button === 1,
   ctrlKey = has("mac") ? evt.metaKey : evt.ctrlKey;
if (middleButton || (leftButton && ctrlKey))
{
   // Internal logic
}
```

The final result is testing whether the internal logic has been executed. Since upgrading, it no longer is when running the intern test on a linux vagrant machine in both FF and Chrome (although it works fine when tested manually in-browser on both a Linux machine and a Mac). Upgrading to 3.2.1 also does not fix the issue.

**Comment by rodneyrehm** [comment]
It seems that `pressKeys()` is using [`/session/:sessionId/element/:id/value`](https://github.com/SeleniumHQ/selenium/wiki/JsonWireProtocol#sessionsessionidelementidvalue) instead of [/session/:sessionId/keys](https://github.com/SeleniumHQ/selenium/wiki/JsonWireProtocol#sessionsessionidkeys). The former doesn't retain modifier state, thus your click does not see the CTRL anymore, because the key has already been released.

The test for [supportsSessionKeys](https://github.com/theintern/leadfoot/blob/79ca6f48e5a30457d82dfd47eb56235ba3c191f0/Server.js#L1025-L1026) calls [pressKeys()](https://github.com/theintern/leadfoot/blob/79ca6f48e5a30457d82dfd47eb56235ba3c191f0/Session.js#L1311-L1329), which itself relies on `supportsSessionKeys`. By simply reading the code, I'd assume this is an oversight?!

**Comment by MartinDoyleUK** [comment]
Just to clarify, @rodneyrehm, are you commenting on how to fix the problem in my codebase, or in the Intern/Leadfoot codebase? I'm assuming the latter?

**Comment by rodneyrehm** [comment]
The latter. If you want to force the old behavior, you could try overwriting the capability:

```js
bdd.it('should force capabilities.supportsSessionKeys', function() {
  return this.remote
    .then(function() {
      this.session.capabilities.supportsSessionKeys = true;
    })
    // …
});
```

**Comment by MartinDoyleUK** [comment]
Thank you. I'm not sure that's really feasible though. I'll roll back and freeze our Intern version on 3.1.1 until this is fixed.

**Comment by MartinDoyleUK** [comment]
@jason0x43, I know you're very busy, but might you have a moment to suggest how long it'll be until you can have a look at this? Is it likely to be days or weeks?

**Comment by jason0x43** [comment]
@rodneyrehm pressKeys should be using `/keys` if it's available (which it should generally be), but since the `supportsSessionKeys` check was not properly implemented, `/value` was being used in cases it shouldn't be. I have this fixed locally, and I will hopefully have an update out this week that fixes this issue.

**Comment by MartinDoyleUK** [comment]
Thank you very much, @jason0x43 :)

**Comment by seriema** [comment]
@jason0x43 was this fix the one in https://github.com/theintern/intern/releases/tag/3.2.3 ?

We still have the same issue with that version (`ctrlKey` is false). In 3.1.0 we can get the correct KeyboardEvent modifiers when we set `requireWindowFocus: true`, but Intern hangs in its capability testing phase as the mouse gets stuck outside the viewport. Setting `fixSessionCapabilities: false` bypasses that issue. But in 3.2.3 the ctrlKey is wrong, even with those settings.

**Comment by jason0x43** [comment]
@seriema Yes, it was. I really need to be better about closing issues. :)

What browser are you targeting?

**Comment by seriema** [comment]
@jason0x43 IE11, on Windows 10.

**Comment by jason0x43** [comment]
@seriema, thanks for the test case, it does reproduce the issue for me. After looking into this a bit more, it seems like it's [a known issue with IE 11 and IEDriverServer](https://github.com/seleniumhq/selenium-google-code-issue-archive/issues/7571). Hopefully it's something Leadfoot can work around; if nothing else we can disable some/all feature tests for IE11 (like we do for Safari), but that won't protect a user from shooting themselves in the foot.

**Comment by seriema** [comment]
@jason0x43 Thanks. I had seen a similar issue, and the workaround [in this comment](https://github.com/SeleniumHQ/selenium/issues/978#issuecomment-179244121) is the one I mentioned above (it works in 3.1.x but not newer versions). Although sometimes one of our tests fail and trying again works, so it's not a stable workaround. Maybe knowing how `requireWindowFocus` and `fixSessionCapabilities` affects the kb modifiers will help Leadfoot work around it?
